### PR TITLE
[supervisor] Be more robust against a too aggressive reaper

### PR DESCRIPTION
**Pull request by csweichel**
## Description
The child process reaper in supervisor is too aggressive at times, especially during shutdown. This occasionally breaks headless workspaces, e.g. image builds. This PR makes the error reporting a bit more informative and does not report `waitid` failures for headless tasks.

## Related Issue(s)

/approve no-issue

## How to test
```bash
cd test
go test -v -run TestBaseImageBuild ./... -count 1
```

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)**
- **[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by akosyakov — APPROVED** [review]
/approve no-issue

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: bc66f4786b0b22f6044f506afa321ac386f514e2</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5566#pullrequestreview-747718117" title="Approved">akosyakov</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **csweichel** after the PR has been reviewed.

Associated issue requirement bypassed by: *<a href="https://github.com/gitpod-io/gitpod/pull/5566#pullrequestreview-747718117" title="Approved">akosyakov</a>*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/content-service/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/content-service/OWNERS)**
- ~~[components/supervisor/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/supervisor/OWNERS)~~ [akosyakov]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
